### ISSUE TITLE
Psalm fails to report more than 1 call to missing methods

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#345]
It appears Psalm only reports calls to missing methods once per statement block. Here we have calls to 4 missing methods on 2 different classes, however psalm check results in a single error:
```php
<?php
class A {}
class B {}

function work_work_zug_zug_dabu(A $p, B $p2): void
{
    $p->missingMethod1();
    $p->missingMethod2();
    $p2->missingMethod2();
    $p2->missingMethod1();
}
```
```
ERROR: UndefinedMethod - qq.php:7:5 - Method A::missingmethod1 does not exist
    $p->missingMethod1();
```
If I wrap every call into `if (true) { ... }` construct, all 4 calls are reported properly.

**Comment by muglug** [comment]
This is by design. If Psalm encounters an un-suppressed issue in a group of statements, it will generally just stop looking for more errors.

What this avoids is a situation where the method call is on the RHS of an assignment, in which case the fallback is an assignment to `mixed`:

```php
$a = $b->undefinedMethod();
$a->foo();
```

If you’re in `totallyTyped` mode you’ll get a ton of new issues complaining about anything that touches `$a`. It seems preferable to stop on the obvious issue, even if it means Psalm has to be run multiple times to catch all the errors.

**Comment by weirdan** [comment]
Makes sense. However it may mask some errors in cases like this:
```php
class A {
    public function isAvailable(): bool
    {
        return false; // we know that knownMissing is not yet there
    }
}
class B {}

function work_work_zug_zug_dabu(A $p, B $p2): void
{
    $p->isAvailable() and $p->knownMissing();
    $p->unknownMissing();
    $p2->unknownMissing();
    $p2->unknownMissing();
}
```
Guess I'll have to review and suppress all expected issues then.
